acceptance/tests/restart_host_run_puppet.rb

```diff
@@ -48,7 +48,14 @@
 
   step "restart each agent" do
     applicable_agents.each do |agent|
-      agent.reboot
+      # Work-around reboot hanging on reply (BKR-971)
+      begin
+        Timeout.timeout(5) do
+          agent.reboot
+        end
```

Comment:
This may not work as expected... If agent.reboot calls into a native syscall, then ruby won't be able to interrupt the thread. It might be better to just sleep for 5 seconds and then call close explicitly.

http://www.mikeperham.com/2015/05/08/timeout-rubys-most-dangerous-api/
http://jvns.ca/blog/2015/11/27/why-rubys-timeout-is-dangerous-and-thread-dot-raise-is-terrifying/